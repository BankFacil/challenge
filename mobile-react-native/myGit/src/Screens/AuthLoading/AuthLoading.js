import React from 'react';
import {
  ActivityIndicator,
  AsyncStorage,
  StatusBar,
  StyleSheet,
  View,
} from 'react-native';
import { connect } from 'react-redux';

import { ROUTES } from '../../Routes/Routes';

import {
  Wrapper,
} from './AuthLoading.style';

export class AuthLoading extends React.Component {
  componentDidMount() {
    const { isLogged, navigation } = this.props;
    if (navigation) {
      navigation.navigate(isLogged ? ROUTES.MAIN : ROUTES.AUTH);
    }
  }

  render() {
    return (
      <Wrapper>
        <ActivityIndicator />
        <StatusBar barStyle="default" />
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  isLogged: !!(state.user.user || {}).credential
});

export default connect(
  mapStateToProps,
)(AuthLoading);
