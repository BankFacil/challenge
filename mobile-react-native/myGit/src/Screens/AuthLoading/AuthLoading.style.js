import styled from 'styled-components/native';

import { Layout } from '../../Styleguide';

export const Wrapper = styled.View`
  ${Layout.Global.wrapper};
`;
