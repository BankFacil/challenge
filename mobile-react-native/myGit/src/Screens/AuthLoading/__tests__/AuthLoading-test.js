/**
 * @format
 */

import 'react-native';
import React from 'react';
import { AuthLoading } from '../AuthLoading';
import renderer from 'react-test-renderer';

it('renders AuthLoading correctly', () => {
  const component = renderer.create(<AuthLoading />).toJSON();
  expect(component).toMatchSnapshot();
});
