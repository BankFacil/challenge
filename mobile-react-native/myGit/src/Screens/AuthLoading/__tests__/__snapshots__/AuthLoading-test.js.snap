// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AuthLoading correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 20,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
